crates/bitcoin-da/src/helpers/test_utils.rs

```diff
@@ -23,26 +30,50 @@ pub(crate) fn get_mock_txs() -> Vec<Transaction> {
         .collect()
 }
 
-pub(crate) fn get_blob_with_sender(tx: &Transaction) -> Result<BlobWithSender, ParserError> {
-    let tx = tx.clone();
-
-    let parsed_transaction = parse_batch_proof_transaction(&tx)?;
-
-    let (blob, public_key) = match parsed_transaction {
-        super::parsers::ParsedBatchProofTransaction::SequencerCommitment(seq_com) => {
-            (seq_com.body, seq_com.public_key)
+pub(crate) fn get_blob_with_sender(
+    tx: &Transaction,
+    ty: MockData,
+) -> Result<BlobWithSender, ParserError> {
+    let (blob, public_key, hash) = match ty {
+        MockData::BatchProof => {
+            let parsed_tx = parse_batch_proof_transaction(tx)?;
+            match parsed_tx {
+                super::parsers::ParsedBatchProofTransaction::SequencerCommitment(seq_com) => {
+                    let hash = seq_com
+                        .get_sig_verified_hash()
+                        .expect("Invalid sighash on commitment");
+                    (seq_com.body, seq_com.public_key, hash)
+                }
+            }
+        }
+        MockData::LightClientProof => {
+            let parsed_tx = parse_light_client_transaction(tx)?;
+            match parsed_tx {
+                super::parsers::ParsedLightClientTransaction::Complete(complete) => {
+                    let hash = complete
+                        .get_sig_verified_hash()
+                        .expect("Invalid sighash on complete zk proof");
+                    let blob = decompress_blob(&complete.body);
+                    (blob, complete.public_key, hash)
+                }
+                super::parsers::ParsedLightClientTransaction::Aggregate(aggregate) => {
+                    let hash = aggregate
+                        .get_sig_verified_hash()
+                        .expect("Invalid sighash on aggregate zk proof");
+                    (aggregate.body, aggregate.public_key, hash)
+                }
+                _ => unimplemented!(),
```

Comment:
why unimplemented? we can get chunk in a block

Reply:
we should ignore chunked transactions in here as this function is used for retrieving relevant prefixed transactions, and requesting chunks is later going to be performed on da service.

this also means that we can't write a unit test currently for extract_relevant_zk_proof as it requires local running regtest bitcoin. I left it as ignored for now to have a reference when we carry the regtest needing tests to bitcoin-e2e.